### PR TITLE
Allow subsequent dispatches to work after store throws error

### PR DESCRIPTION
**Pull request by jordangarcia**

### CODE REVIEW

**Comment by balloob** [comment]
Sorry for missing this case!

**Comment by bhamodi** [comment]
Oops! Good find @jordangarcia. LGTM. :+1:

**Comment by jordangarcia** [comment]
@balloob mind giving this a review?

**Comment by jordangarcia** [comment]
Ran this against optimizely's test suite of 1500 tests, no regressions.  Going to merge and cut `1.1`.

File: src/reactor.js [hunk 1]
```diff
@@ -299,7 +313,12 @@ class Reactor {
       if (this.__batchDispatchCount > 0) {
         // set to true to catch if dispatch called from observer
         this.__isDispatching = true
-        this.__notify()
+        try {
+          this.__notify()
+        } catch (e) {
+          this.__isDispatching = false
```
Comment (balloob) [inline src/reactor.js]: Very minor nit picky: this is a perfect use case to use `finally`.

**Comment by balloob** [comment]
Sorry, I saw the request after you've merged. It looks good to me!

**Pull request merged**